src/System.Private.CoreLib/shared/System/Type.cs

```diff
@@ -281,8 +281,9 @@ public static TypeCode GetTypeCode(Type type)
         }
         protected virtual TypeCode GetTypeCodeImpl()
         {
-            if (this != UnderlyingSystemType && UnderlyingSystemType != null)
-                return Type.GetTypeCode(UnderlyingSystemType);
+            Type systemType = UnderlyingSystemType;
+            if (this != systemType && systemType != null)
```

Comment:
Does ` != ` call an operator here?

Maybe something like:
```cs
if ((object)this != systemType && (object)systemType != null)
```

Reply:
It calls the operator and it should continue to call the operator. This works on foreign type objects that may not have referential equality.

The comparison with null may use `is` or `ReferenceEquals`, but it does not really matter because of JIT optimizations will take care of it.

Reply:
I did not realize that `UnderlyingSystemType` could return a foreign type object. Makes sense then.